Documentation/project-docs/writing-tests.md

```diff
@@ -0,0 +1,93 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
```

Comment:
> I think the only reason this app exists is process-level isolation for reliable Diagnostics collection and shouldn't be used for other purposes.

Not true.  It's very useful for:
- Being able to test things that require environment changes, e.g. how environment variables impact the app
- Being able to isolate changes to statics; you say this can be achieved as well by disabling parallelization, but a) that assumes you can reliably reset static state which isn't always the case, and b) disabling parallelization has a real impact on dev productivity.
- Being able to verify that things that should crash do crash
- Being able to verify that things don't depend on state that's been configured previously, e.g. that some code you're calling doesn't require that some previous related code ran (e.g. that you can deserialize some state without it having previously been serialized in the same process)
- Being able to test cross-process support for various things, e.g. cross-process synchronization, cross-process memory-mapped files, that file locking works correctly cross-process, etc.
Etc.